### PR TITLE
Handle stop/abort websocket signals

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: neptune/internal/execution/execution_context.py [hunk 1]
```diff
@@ -137,10 +136,14 @@ def _run_aborting_thread(self, abort_callback):
...
 
-        websocket_factory = ReconnectingWebsocketFactory(
-            backend=self._backend,
+        websocket_factory = self._backend.websockets_factory(
+            # pylint: disable=protected-access
+            project_uuid=self._experiment._project.internal_id,
             experiment_id=self._experiment.internal_id
         )
+        if not websocket_factory:
+            return
```
Comment (HubertJaworski) [inline neptune/internal/execution/execution_context.py:145]: maybe this should not return and instead do `if websocket_factory -> AbortingThread()`

File: neptune/new/attributes/constants.py [hunk 2]
```diff
@@ -36,3 +36,4 @@
 SYSTEM_NAME_ATTRIBUTE_PATH = f'{SYSTEM_ATTRIBUTE_SPACE}name'
 SYSTEM_STATE_ATTRIBUTE_PATH = f'{SYSTEM_ATTRIBUTE_SPACE}state'
 SYSTEM_TAGS_ATTRIBUTE_PATH = f'{SYSTEM_ATTRIBUTE_SPACE}tags'
+SYSTEM_FAILED_ATTRIBUTE_PATH = f'{SYSTEM_ATTRIBUTE_SPACE}failed'
```
Comment (HubertJaworski) [inline neptune/new/attributes/constants.py:39]: Please rebase to https://github.com/neptune-ai/neptune-client/pull/521 where it's added

File: neptune/new/internal/websockets/websocket_signals_background_job.py [hunk 3]
```diff
@@ -67,10 +71,37 @@ def work(self) -> None:
...
+                if not msg_type:
+                    click.echo(f"Malformed websocket signal: missing type", err=True)
+                    return
+                if not isinstance(msg_type, str):
+                    click.echo(f"Malformed websocket signal: type is {type(msg_type)}, should be str", err=True)
+                    return
+                if not isinstance(msg_body, dict):
+                    click.echo(f"Malformed websocket signal: body is {type(msg_body)}, should be dict", err=True)
+                    return
+                if msg_type.lower() == 'neptune/stop':
```
Comment (HubertJaworski) [inline neptune/new/internal/websockets/websocket_signals_background_job.py:93]: Externalize to some constants maybe?

File: neptune/new/internal/websockets/websocket_signals_background_job.py [hunk 4]
```diff
@@ -67,10 +71,37 @@ def work(self) -> None:
...
+                json_msg = json.loads(msg)
+                msg_type = json_msg.get("type")
+                msg_body = json_msg.get("body") or dict()
+                if not msg_type:
+                    click.echo(f"Malformed websocket signal: missing type", err=True)
+                    return
+                if not isinstance(msg_type, str):
+                    click.echo(f"Malformed websocket signal: type is {type(msg_type)}, should be str", err=True)
+                    return
+                if not isinstance(msg_body, dict):
```
Comment (HubertJaworski) [inline neptune/new/internal/websockets/websocket_signals_background_job.py:90]: body is object in API. We should not require it to be dict in client

**Review by HubertJaworski — CHANGES_REQUESTED** [review]

File: neptune/internal/execution/execution_context.py (hunk 1 above)
Comment (aniezurawski) [reply neptune/internal/execution/execution_context.py:145]: I don't get this comment.

File: neptune/new/attributes/constants.py (hunk 2 above)
Comment (aniezurawski) [reply neptune/new/attributes/constants.py:39]: It has not been merged yet.

**Review by HubertJaworski — APPROVED** [review]

**Pull request merged**